### PR TITLE
fix: Don't allow micronaut as app name

### PR DESCRIPTION
**Pull request by sdelamo**
Close: https://github.com/micronaut-projects/micronaut-starter/issues/352

See: https://github.com/micronaut-projects/micronaut-starter/issues/401

### CODE REVIEW

File: starter-api/src/main/java/io/micronaut/starter/api/create/AbstractCreateController.java [hunk 1]
```diff
@@ -80,6 +81,9 @@ protected AbstractCreateController(
             @Nullable Language lang,
             @Nullable JdkVersion javaVersion,
             String userAgent) {
+        if (name.equalsIgnoreCase(MICRONAUT)) {
+            throw new HttpStatusException(HttpStatus.BAD_REQUEST, "project name cannot be " + name);
+        }
```
Comment (ilopmar) [inline starter-api/src/main/java/io/micronaut/starter/api/create/AbstractCreateController.java:86]: Wouldn't it be better to check the project name directly in `NameUtils.parse(name)`?

https://github.com/micronaut-projects/micronaut-starter/pull/403/files#diff-f5c89fa2c105c0ad496c0386652c7389L84-L88

File: starter-cli/src/main/java/io/micronaut/starter/cli/command/CreateCommand.java [hunk 2]
```diff
@@ -105,6 +106,10 @@ public Integer call() throws Exception {
             throw new IllegalArgumentException("Specify an application name or use --inplace to create an application in the current directory");
         }
 
+        if (name.equalsIgnoreCase(MICRONAUT)) {
+            throw new CommandLine.ParameterException(this.spec.commandLine(), MICRONAUT + " is not a valid application name\n");
+        }
+
```
Comment (ilopmar) [inline starter-cli/src/main/java/io/micronaut/starter/cli/command/CreateCommand.java:112]: Same comment as before.

File: starter-core/src/main/java/io/micronaut/starter/application/Project.java [hunk 3]
```diff
@@ -20,26 +20,30 @@
...
-public class Project {
+public class Project extends ProjectIdentifier {
 
-    private final String packageName;
+    public static final String PACKAGE_NAME = "packageName";
+    public static final String PACKAGE_PATH = "packagePath";
+    public static final String CLASS_NAME = "className";
+    public static final String NATURAL_NAME = "naturalName";
+    public static final String PROPERTY_NAME = "propertyName";
+    public static final String NAME = "name";
```
Comment (ilopmar) [inline starter-core/src/main/java/io/micronaut/starter/application/Project.java:30]: I think this can be `private` or at least package protected.

Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/application/Project.java]: I changed them to `private`.

**Pull request merged**